src/Compilers/CSharp/Test/Semantic/Semantics/CollectionLiteralTests.cs

```diff
@@ -1034,1026 +1290,2634 @@ static void Main()
                     }
                 }
                 """;
-            var verifier = CompileAndVerify(new[] { source, GetCollectionExtensions() }, expectedOutput: "[], [null, 2], ");
-            verifier.VerifyIL("Program.CreateEmpty<T, U>", """
-                {
-                  // Code size        6 (0x6)
-                  .maxstack  1
-                  IL_0000:  call       "T System.Activator.CreateInstance<T>()"
-                  IL_0005:  ret
-                }
-                """);
-            verifier.VerifyIL("Program.Create<T, U>", """
-                {
-                  // Code size       36 (0x24)
-                  .maxstack  2
-                  .locals init (T V_0)
-                  IL_0000:  call       "T System.Activator.CreateInstance<T>()"
-                  IL_0005:  stloc.0
-                  IL_0006:  ldloca.s   V_0
-                  IL_0008:  ldarg.0
-                  IL_0009:  constrained. "T"
-                  IL_000f:  callvirt   "void I<U>.Add(U)"
-                  IL_0014:  ldloca.s   V_0
-                  IL_0016:  ldarg.1
-                  IL_0017:  constrained. "T"
-                  IL_001d:  callvirt   "void I<U>.Add(U)"
-                  IL_0022:  ldloc.0
-                  IL_0023:  ret
-                }
-                """);
+            var comp = CreateCompilation(source);
+            // PROTOTYPE: ERR_AnonMethGrpInForEach is misleading since there is no 'foreach' in source.
+            comp.VerifyEmitDiagnostics(
+                // (6,21): error CS0446: Foreach cannot operate on a 'collection literals'. Did you intend to invoke the 'collection literals'?
+                //         a = [..a, ..[]];
+                Diagnostic(ErrorCode.ERR_AnonMethGrpInForEach, "[]").WithArguments("collection literals").WithLocation(6, 21));
         }
 
         [Fact]
-        public void TypeParameter_02()
+        public void SpreadElement_06()
         {
             string source = """
-                using System.Collections;
-                using System.Collections.Generic;
-                interface I<T> : IEnumerable<T>
-                {
-                    void Add(T t);
-                }
-                struct S<T> : I<T>
-                {
-                    public void Add(T t) { }
-                    IEnumerator<T> IEnumerable<T>.GetEnumerator() => throw null;
-                    IEnumerator IEnumerable.GetEnumerator() => throw null;
-                }
                 class Program
                 {
-                    static T Create1<T, U>() where T : struct, I<U> => [];
-                    static T? Create2<T, U>() where T : struct, I<U> => [];
+                    static string[] Append(string a, string b, bool c)
+                    {
+                        return [a, b, .. c ? [null] : []];
+                    }
                 }
                 """;
+            // PROTOTYPE: Should we infer List<string> rather than reporting an error?
```

Comment:
I would expect us to target-type the spread element, and the collection literals `[null]` and `[]` within, to `string[]`.

Reply:
Target-typing the spread element sounds reasonable. Is there a reason to choose `string[]` over another constructible collection type with `string` element type?